### ISSUE TITLE
suggesion: take anchors (id=) into the account during indexing HTML

### ISSUE THREAD

**Issue by karlicoss** [jameslittle230/stork#159]
Was nice meeting you at the Fission demo! :)

A nice feature to have for a static site indexer is taking HTML anchors into the account. It would allow to suggest more specific results and jump straight to the occurrence (by using the `id` value)

- Roadmap: perhaps it's related to
  - "Build and publish some integrations with other projects" as many are generating anchors. So by running against generated HTML it could already benefit from it while being agnostic of specific site generator
  - "Index different types of content": jumping at a specific place in code/subtitle might be desirable. And again, often HTML versions already have some anchors, so by taking them into the account might end up supporting some of these formats collaterally.

I think some basic version could be relatively easy to implement with a small amount of code -- I implemented it for my [exobrain](https://beepb00p.xyz/exobrain/) in about 30 LOC: https://github.com/karlicoss/beepb00p/blob/20f47da1779edf8c05869ab6618be32f9d203a0a/exobrain/src/search/makeindex.py#L13-L42  

The basic idea of my algorithm is

- traverse the DOM recursively
- for each anchor we encounter, we say that all elements in it's DOM subtree correspond to its own 'document'
- unless some of its children are also anchors, in which case they get priority to 'own' their subtrees

So basically, each piece of text 'belongs' to the closest parent anchor. Perhaps a slight improvement could also be taking into the account siblings (until the next anchor occurs). This would help with layouts like

```
<div id="heading1">Heading 1</div>
<div> .... body1 .... </div>
<div id="heading2">Heading 2<div>
<div> .... body2 .... </div>
```

What do you think? :)

**Comment by jameslittle230** [comment]
I love it!! Definitely worth adding to the HTML indexer.
